### DISCUSSION TITLE
How to update apt packages in an air gapped environment

### DISCUSSION THREAD

**Discussion by cospeedster** [renovatebot/renovate#30477] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure DevOps Server (pipeline container job)

### Please tell us more about your question or problem

Hi all,
I am wondering if it's possible to update apt package versions in an air gapped environment?
We've a local mirrored or cached apt repositoy and cannot use the repology source (#3722) because we've neither internet access nor proxy.

**Comment by rarkins** [comment, marked as answer]
Not possible currently, may be possible after https://github.com/renovatebot/renovate/pull/30071

**Reply by cospeedster** [reply]
Thanks for your reply, this aswers my question :-)

**Reply by cospeedster** [reply]
I'll try the new [deb datasource](https://docs.renovatebot.com/modules/datasource/deb/) which was introduced in release [38.57.0](https://github.com/renovatebot/renovate/releases/tag/38.57.0).
